### PR TITLE
fix: support attributes in TargetFrameworks

### PR DESCRIPTION
**Pull request by admons**
Today we don't support attributes on the TargetFrameworks property in .csproj files
This change adds this support

- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/dotnet-deps-parser/blob/master/CONTRIBUTING.md) rules
- [ ] Reviewed by Snyk internal team

### CODE REVIEW

**Comment by orsagie** [comment]
What happens if there are two <TargetFramewoks/> tags? each with a different condition?
Can you add a test case for that?

**Comment by admons** [comment]
@orsagie 
I searched for and didn't saw any reference for more than one TargetFrameworks. 
Doesn't really make sense to have attributes here, and I'm pretty sure it's not respected. 
The reason for this change is just not to fail here

File: test/lib/target-frameworks.test.ts [hunk 1]
```diff
@@ -22,16 +23,25 @@ test('.Net F# project target framework extracted as expected', async (t) => {
...
 
 test('.Net .csproj simple project target framework extracted as expected', async (t) => {
   const targetFrameworks = await extractTargetFrameworksFromFiles(
     `${__dirname}/../fixtures/dotnet-core-simple-project`,
     'simple-project.csproj');
   t.deepEqual(targetFrameworks, ['netcoreapp1.1', 'netcoreapp1.2'], 'targetFramework array is as expected');
 });
+*/
+
+test('.Net .csproj simple project complex target framework extracted as expected', async (t) => {
```
Comment (jan-stehlik) [inline test/lib/target-frameworks.test.ts:38]: Nitpick: shall we be more explicit to what we mean by complex?

```suggestion
test('.Net .csproj simple project with two target frameworks extracted as expected', async (t) => {
```

**Review by jan-stehlik — COMMENTED** [review]
Hey @admons , what's the best way to test this please? Should I 
1. run dotnet-deps
2. link dotnet-deps-parser to dotnet-deps
3. ??
Would you be able to help please?

File: test/lib/target-frameworks.test.ts (hunk 1 above)
Comment (admons) [reply test/lib/target-frameworks.test.ts:38]: 👍

File: lib/parsers/index.ts [hunk 2]
```diff
@@ -444,6 +444,13 @@ export function getTargetFrameworksFromProjectFile(manifestFile) {
   return _uniq(targetFrameworksResult);
 }
 
+function getTargetFrameworks(item: string | any) {
+  if (typeof item === 'object' && item.hasOwnProperty('_')) {
+    item = item._;
```
[the remark this answers is not in the document]
Comment (admons) [reply lib/parsers/index.ts:449]: `item == "something"` for `<TargetFrameworks>something</TargetFrameworks>`
`item._ == "something"` for `<TargetFrameworks someAttribute="someValue">something</TargetFrameworks>`

**Review by aviadhahami — APPROVED** [review]

**Comment by orsagie** [comment]
> @orsagie
> I searched for and didn't saw any reference for more than one TargetFrameworks.
> Doesn't really make sense to have attributes here, and I'm pretty sure it's not respected.
> The reason for this change is just not to fail here

This is exactly the scenario we saw fail
```<Project Sdk="Microsoft.NET.Sdk">
<PropertyGroup>
<OutputType>Exe</OutputType>
<TargetFrameworks Condition="$(OS) == 'Windows_NT'">net472;net5.0</TargetFrameworks>
<TargetFramework Condition="$(OS) != 'Windows_NT'">net5.0</TargetFramework>
<RootNamespace>snyk_repro</RootNamespace>
</PropertyGroup>
</Project>```

**Pull request merged**